xcube/api/gen/gen.py

```diff
@@ -150,6 +151,15 @@ def monitor(*args):
     return output_path, status
 
 
+def _check_time_valid(time_range, input_file):
+    if time_range[0] > time_range[1]:
+        print(f"illegal time range: file {input_file}:"
```

Comment:
Don't print text directly, either use Python logging or warning modules.
Why printing at all? This should be the exception's message string.